internal/server/resp/server.go

```diff
@@ -0,0 +1,221 @@
+package resp
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/dicedb/dice/config"
+	"github.com/dicedb/dice/internal/clientio/iohandler/netconn"
+	respparser "github.com/dicedb/dice/internal/clientio/requestparser/resp"
+	"github.com/dicedb/dice/internal/ops"
+	"github.com/dicedb/dice/internal/shard"
+	"github.com/dicedb/dice/internal/worker"
+)
+
+var (
+	workerCounter uint64
+	startTime     = time.Now().UnixNano() / int64(time.Millisecond)
+)
+
+var (
+	ErrInvalidIPAddress = errors.New("invalid IP address")
+)
+
+const (
+	DefaultConnBacklogSize = 128
+)
+
+type Server struct {
+	Host            string
+	Port            int
+	serverFD        int
+	connBacklogSize int
+	wm              *worker.WorkerManager
+	sm              *shard.ShardManager
+	globalErrorChan chan error
+	logger          *slog.Logger
+}
+
+func NewServer(sm *shard.ShardManager, wm *worker.WorkerManager, gec chan error, l *slog.Logger) (*Server, error) {
+	return &Server{
+		Host:            config.DiceConfig.Server.Addr,
+		Port:            config.DiceConfig.Server.Port,
+		connBacklogSize: DefaultConnBacklogSize,
+		wm:              wm,
+		sm:              sm,
+		globalErrorChan: gec,
+		logger:          l,
+	}, nil
```

Comment:
What are the scenarios in which RESP server initialization will lead to error ?

Reply:
Thanks for catching. removed error.